### DISCUSSION TITLE
H265 coral and audio encoding? no cpu

### DISCUSSION THREAD

**Discussion by ukscarface** [blakeblackshear/frigate#4702] [General]
Help?? i have new cameras installed and are ONLY H265 I cant get them to work without CPU processing for ffmpeg on the fly meaning i can only run few cameras or I only have the green screen, there must be a way around this

**Comment by NickM-27** [comment]
What do you mean by CPU processing on the fly? Whether a camera is h264 or h265 you should be using hardware acceleration

**Reply by ukscarface** [reply]
i use no hardware acceleration as does not work with h265

**Reply by NickM-27** [reply]
What device are you running on? Besides the RPi, all hwaccel types support h265

**Comment by ukscarface** [comment]
BackDoor: 
    ffmpeg:
      output_args:
        record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -tag:v hvc1 -bsf:v hevc_mp4toannexb -c:a aac
        rtmp: -c:v copy -c:a aac -f flv
        
      inputs:
        - path: rtsp://ipaddress:554/user=admin_password=password_channel=1_stream=0.sdp?
          roles:
            - detect
            - record
            - rtmp
    rtmp:
      enabled: false
    detect:
      width: 3840 
      height: 2160 
      fps: 7

**Reply by NickM-27** [reply]
We highly recommend not running detect on the high res stream, you should be using a sub stream. https://docs.frigate.video/guides/camera_setup#choosing-a-detect-resolution

**Reply by ukscarface** [reply]
yes i would do this but the sub steam also is h265 but it wont load

**Reply by NickM-27** [reply]
I'd recommend setting it that way then create an actual issue here and provide all the requested information, that way I can understand what is going on and help

**Reply by ukscarface** [reply]
how do i add hwaccel to these streams also?

**Reply by NickM-27** [reply]
What device are you running frigate on? and what CPU is it

**Reply by ukscarface** [reply]
NUC and im using home assistant addon full access works awesome just not with H265

**Reply by NickM-27** [reply]
It can definitely have hwaccel for h265, just use `hevc_qsv` instead of `h264_qsv` as the hardware accel type

**Reply by ukscarface** [reply]
also no live audio on the H265 stream but fine with the others Intel CPU

**Reply by ukscarface** [reply]
> It can definitely have hwaccel for h265, just use `hevc_qsv` instead of `h264_qsv` as the hardware accel type

i have added the hwaccel as you have said like this but now green screen

  Garage: 
    ffmpeg:
      hwaccel_args: hevc_qsv
      output_args:
        record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -tag:v hvc1 -bsf:v hevc_mp4toannexb -c:a aac
        rtmp: -c:v copy -c:a aac -f flv
        
      inputs:
        - path: rtsp://192.168.4.63:554/user=admin_password=password1_channel=1_stream=0.sdp?

**Reply by NickM-27** [reply]
Please make an issue, without seeing your config I can't help.

**Reply by NickM-27** [reply]
> i have added the hwaccel as you have said like this but now green screen

That is not what I said, please read the docs https://docs.frigate.video/configuration/hardware_acceleration#intel-based-cpus-10th-generation-via-quicksync-1 I simply said use `hevc_qsv` instead of `h264_qsv`

**Reply by ukscarface** [reply]
ok sorry i also have tried the wey from the frigate site

ffmpeg:
  hwaccel_args: -c:v h264_qsv

Green screen argggggg its close i can feel it!

**Reply by NickM-27** [reply]
Please slow down and read what I'm saying, you should use 

ffmpeg:
hwaccel_args: -c:v hevc_qsv

As the args for h265 stream

**Reply by ukscarface** [reply]
greatly appreciate your time but again soon as this is added the stream goes green

**Reply by ukscarface** [reply]
https://docs.frigate.video/configuration/camera_specific

Annke C800 is what im using

**Reply by NickM-27** [reply]
> soon as this is added the stream goes green

this is not helpful, green screen means nothing. You need to provide logs or I won't be able to help. Again, please just create an actual issue and answer all the questions
